src/conf_mode/vpp.py

```diff
@@ -131,32 +131,44 @@ def verify(config):
         return None
 
     if 'interface' not in config:
-        raise ConfigError(f'"interface" is required but not set!')
+        raise ConfigError('"interface" is required but not set!')
 
     if 'cpu' in config:
-        if 'corelist_workers' in config['cpu'] and 'main_core' not in config['cpu']:
-            raise ConfigError(f'"cpu main-core" is required but not set!')
+        if 'corelist_workers' in config['cpu'] and 'main_core' not in config[
+                'cpu']:
+            raise ConfigError('"cpu main-core" is required but not set!')
 
-    memory = psutil.virtual_memory()
-    memory_total = round(memory.total / (1024 ** 3), 2)
-    if memory_total < MIN_TOTAL_MEMORY:
+    memory_available: int = virtual_memory().available
+    if memory_available < MIN_AVAILABLE_MEMORY:
         raise ConfigError(
-            f'Not enough installed memory {memory_total}GB! '
-            f'The minimum required memory is {MIN_TOTAL_MEMORY}GB.'
-        )
+            'Not enough free memory to start VPP:\n'
+            f'available: {round(memory_available / 1024**3, 1)}GB\n'
+            f'required: {round(MIN_AVAILABLE_MEMORY / 1024**3, 1)}GB')
 
 
 def generate(config):
     if not config or (len(config) == 1 and 'removed_ifaces' in config):
         # Remove old config and return
         service_conf.unlink(missing_ok=True)
-        if os.path.isfile(sysctl_vpp):
-            os.unlink(sysctl_vpp)
         return None
 
     render(service_conf, 'vpp/startup.conf.j2', config)
     render(systemd_override, 'vpp/override.conf.j2', config)
-    render(sysctl_vpp, 'vpp/sysctl.conf.j2', config)
+
+    # apply sysctl settings
```

Comment:
Maybe add some comments about the reasoning behind these values? Since I haven't been following the VPP development, they now seem magical to me, for example.

Reply:
They are just default values from https://github.com/FDio/vpp/blob/v23.06/src/vpp/conf/80-vpp.conf

Reply:
I think it's a good idea to say it explicitly. Their defaults may change in the future, and that reasoning will be forgotten by then, so it's better to document it to avoid self-inflicted puzzles.

Reply:
Totally agree, fixed.